x-pack/plugin/fleet/src/test/java/org/elasticsearch/xpack/fleet/action/GetGlobalCheckpointsActionTests.java

```diff
@@ -0,0 +1,203 @@
+        );
+        final GetGlobalCheckpointsAction.Response response = client().execute(GetGlobalCheckpointsAction.INSTANCE, request).get();
+        long[] expected = new long[shards];
+        for (int i = 0; i < shards; ++i) {
+            expected[i] = -1;
+        }
+        assertArrayEquals(expected, response.globalCheckpoints());
+
+        final int totalDocuments = shards * 3;
+        for (int i = 0; i < totalDocuments; ++i) {
+            client().prepareIndex(indexName).setId(Integer.toString(i)).setSource("{}", XContentType.JSON).get();
+        }
+
+        assertBusy(() -> {
+            final GetGlobalCheckpointsAction.Request request2 = new GetGlobalCheckpointsAction.Request(
+                indexName,
+                false,
+                EMPTY_ARRAY,
+                TEN_SECONDS
+            );
+            final GetGlobalCheckpointsAction.Response response2 = client().execute(GetGlobalCheckpointsAction.INSTANCE, request2).get();
+
+            assertEquals(totalDocuments, Arrays.stream(response2.globalCheckpoints()).map(s -> s + 1).sum());
+
+            final IndicesStatsResponse statsResponse = client().admin().indices().prepareStats(indexName).get();
+            long[] fromStats = Arrays.stream(statsResponse.getShards())
+                .filter(i -> i.getShardRouting().primary())
+                .sorted(Comparator.comparingInt(value -> value.getShardRouting().id()))
+                .mapToLong(s -> s.getSeqNoStats().getGlobalCheckpoint())
+                .toArray();
+            assertArrayEquals(fromStats, response2.globalCheckpoints());
+        });
+    }
+
+    public void testPollGlobalCheckpointAdvancement() throws Exception {
+        String indexName = "test_index";
+        client().admin()
+            .indices()
+            .prepareCreate(indexName)
+            .setSettings(
+                Settings.builder()
+                    .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), Translog.Durability.REQUEST)
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 0)
+            )
+            .get();
+
+        final GetGlobalCheckpointsAction.Request request = new GetGlobalCheckpointsAction.Request(
+            indexName,
+            false,
+            EMPTY_ARRAY,
+            TEN_SECONDS
+        );
+        final GetGlobalCheckpointsAction.Response response = client().execute(GetGlobalCheckpointsAction.INSTANCE, request).get();
+        assertEquals(-1, response.globalCheckpoints()[0]);
+
+        final int totalDocuments = 30;
+        for (int i = 0; i < totalDocuments; ++i) {
+            client().prepareIndex(indexName).setId(Integer.toString(i)).setSource("{}", XContentType.JSON).execute();
+        }
+
+        final GetGlobalCheckpointsAction.Request request2 = new GetGlobalCheckpointsAction.Request(
+            indexName,
+            true,
+            new long[] { 28 },
+            TEN_SECONDS
```

Comment:
If we extend this to 30 seconds or so, we could assert that it took less than 30 seconds to poll.